Cliqz/Freshtab/News/NewsViewController.swift

```diff
@@ -233,9 +233,12 @@ extension NewsViewController {
 	
 	fileprivate func getExtraNewsIndexPaths() -> [IndexPath] {
 		var indexPaths = [IndexPath]()
-		for i in NewsViewUX.MinNewsCellsCount..<self.dataSource.newsCount() {
-			indexPaths.append(IndexPath(row: i, section: 0))
-		}
+        //self.dataSource.newsCount() < NewsViewUX.MinNewsCellsCount crashes the app
+        if self.dataSource.newsCount() > NewsViewUX.MinNewsCellsCount {
```

Comment:
I would suggest another solution. It needs only to be  changed  "for i in min(NewsViewUX.MinNewsCellsCount, self.dataSource.newsCount() > 0 ? 0 : self.dataSource.newsCount())..<self.dataSource.newsCount() { ... "

It will solve the crash and also in the case if newsCount is less than minNewsCount, but > 0, it will show that news...

Reply:
That's true, I thought it's a general method to get all rows. But then it can be "for i in min(NewsViewUX.MinNewsCellsCount, self.dataSource.newsCount()) ..< self.dataSource.newsCount() ..."  without if statement...

Reply:
You are embedding the if statement in the for statement. It is cleaner if we keep them separated. They are easier to read.

Reply:
My bad. I thought it was the same code as above.

Reply:
I want to leave the code like this. It is easier to reason about. I can easily see what is guaranteed inside the if.

Reply:
@naira-cliqz I think using `if` before the for loop is more readable than embed `min` in the for loop